docs/guides/random-data-for-security.md

```diff
@@ -0,0 +1,196 @@
+
+### Sys::GetRandom
+
+If you are writing code that only runs on Linux or BSD systems with the `getrandom` system call, you can use
+[Sys::GetRandom](https://metacpan.org/pod/Sys::GetRandom), an XS module that calls the function directly.
+To obtain 256-bits (32 bytes) of data using it:
+
+    use Sys::GetRandom qw( random_bytes );
+
+    my $bytes = random_bytes(32);
+
+There is also a pure-Perl version [Sys::GetRandom::PP](https://metacpan.org/pod/Sys::GetRandom::PP) that uses the
+`syscall` function.
+
+Note there are some caveats when using `getrandom` to retrieve more than 256 bytes at a time, as the amount of
+data returned may be less due to interrupts.
+
+## Using Crytographic Strength PRNGs
+
+It's often faster to use random data to seed a cryptographic strength PRNG (CSPRNG) and pull data from that, than to
+repeatedly request random data from the operating system.
+
+It's important to note that CSPRNGs are not "less secure". If the algorithm is considered secure, then a PRNG seeded
+with 256 or more bits is difficult to predict the output of.  If the use case is to generate short-lived tokens then it
+is more than adequate.
+
+### Crypt::PRNG
+
+[Crypt::PRNG](https://metacpan.org/pod/Crypt::PRNG) is part of the [CryptX](https://metacpan.org/dist/CryptX)
+cryptographic toolkit that uses the [libtomcrypt](https://github.com/libtom/libtomcrypt).  It will initialise a
+cryptographic-stregnth PRNG from `/dev/urandom`, and claims to be thread and fork safe.
+
+It also supports several utility methods for returning base-64 strings, URL-safe base-64 strings, random strings or
+strings with custom alphabets, random integers and random floating point numbers.
+
+For example,
+
+    use Crypt::PRNG;
+
+    my $bytes = Crypt::PRNG->new->bytes(32);
+
+will return 246 bits of random data.
+
+### Math::Random::ISAAC
+
+[Math::Random::ISAAC](https://metacpan.org/pod/Math::Random::ISAAC) is a small and very fast CSPRNG
+to generate 32-bit integers and floating point numbers.  There is also an XS-version
+[Math::Random::ISAAC::XS](https://metacpan.org/pod/Math::Random::ISAAC::XS).
+
+One caveat of this module is that it needs to be manually seeded by 256 long integers. This is not difficult:
+
+    use Crypt::URandom qw( urandom );
+    use Math::Random::ISAAC;
+
+    my $rng = Math::Random::ISAAC->new( unpack( "N*", urandom(1024) ) ); # 8192 bits
+
+### Crypt::Bear
+
+[Crypt::Bear](https://metacpan.org/pod/Crypt::Bear) is a Perl wrapper for the [BearSSL](https://bearssl.org/) library.
+It includes CSPRNGs such as [Crypt::Bear::AES_CTR::DRBG](https://metacpan.org/pod/Crypt::Bear::AES_CTR::DRBG) that is
+very fast:
+
+    my $prng = Crypt::Bear::AES_CTR::DRBG->new($seed);
+
+    $prng->system_seed;
+
+    my $bytes - $prng->generate(32);
+
+## Generating Tokens and Passwords
+
+Desires generating raw random data for encryption keys or initialisation vectors, a common need is to generate a
```

Comment:
Desires seems strange as a start.  Maybe "If you need to generate raw..."

Reply:
Hm, I think that's a typo